function_pipe/core/function_pipe.py

```diff
@@ -258,7 +290,10 @@ def __call__(self: FN, *args: tp.Any, **kwargs: tp.Any) -> tp.Any:
         """
         Call the wrapped function with args and kwargs.
         """
-        return self._function(*args, **kwargs)
+        try:
+            return self._function(*args, **kwargs)
+        except BaseException as e:
```

Comment:
Here and elsewhere: are you sure you want to catch all `BaseException`, and not just `Exception`? The goal here is to provide a better user experience, and I expect most user exceptions will be subclasses of `Exception`; if we see a `BaseException`, something really strange might be going on where, perhaps, we want to provide the complete traceback.

Reply:
That's a good point. I will just use `Exception`